v2/paths/snapctl.yaml

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: GPL-3.0
+# SPDX-FileCopyrightText: Canonical Ltd
+
+post:
+  tags:
+    - AuthenticationRequired
+    - Synchronous
+  summary: Run snapctl command
+  description: |-
+    Executes a 'snapctl' command within a given context.
+  operationId: runSnapctl
+  security:
+    - PeerAuth: []
+  requestBody:
+    description: |-
+      The context and arguments for the snapctl command.
+    required: true
+    content:
+      application/json:
+        schema:
+          type: object
+          required:
+            - context-id
+            - args
+          properties:
+            context-id:
+              type: string
+              description: The context ID for this call.
```

Comment:
Should we describe in further detail that the context ID is passed to hooks through the `$SNAP_COOKIE` env var or would this be too much introspection on the internals?

The snapctl command passes it automatically but for hooks that are binaries using this spec, they need to get the context ID themselves.